src/FAudio_gstreamer.c

```diff
@@ -404,7 +436,8 @@ static void FAudio_INTERNAL_DecodeGSTREAMER(
 	if (samples > 0)
 	{
 		if (	!error &&
-			curBlock < voice->src.bufferList->bufferWMA.PacketCount - 1	)
+			curBlock < blockCount - 1 &&
+			availableBytes != 0	)
```

Comment:
`availableBytes != 0` is the other change I've added to prevent decoding from looping infinitely, specifically when the decoder thinks there are more blocks but ran out of data to decode. In my case happened due to block size problems which have now been fixed. Is this still worth keeping?

Reply:
If your stuff still works without the change I would revert it - for this stuff in particular, the less changes we have the better since this file can get pretty sensitive (as you've seen already).